lib/galaxy/authnz/custos_authnz.py

```diff
@@ -42,36 +44,49 @@
 class InvalidAuthnzConfigException(Exception):
     pass
 
-
-class CustosAuthnz(IdentityProvider):
+@dataclass
+class CustosAuthnzConfiguration:
+    provider : str
+    verify_ssl : bool
+    url : str
+    label: str
+    client_id: str
+    client_secret: str
+    require_create_confirmation: str
+    redirect_uri: str
+    ca_bundle: str
+    pkce_support: str
+    extra_params: str
+    authorization_endpoint: str
+    token_endpoint: str
+    end_session_endpoint: str
+    well_known_oidc_config_uri: str
+    iam_client_secret: str
+
+
+class OIDCAuthnzBase(IdentityProvider):
     def __init__(self, provider, oidc_config, oidc_backend_config, idphint=None):
         provider = provider.lower()
-        self.config = {"provider": provider}
-        self.config["verify_ssl"] = oidc_config["VERIFY_SSL"]
-        self.config["url"] = oidc_backend_config["url"]
-        self.config["label"] = oidc_backend_config.get("label", provider.capitalize())
-        self.config["client_id"] = oidc_backend_config["client_id"]
-        self.config["client_secret"] = oidc_backend_config["client_secret"]
-        self.config["require_create_confirmation"] = oidc_backend_config.get(
-            "require_create_confirmation", provider == "custos"
+        self.config = CustosAuthnzConfiguration(provider=provider,
+            verify_ssl=oidc_config["VERIFY_SSL"],
```

Comment:
This needs to be run through the formatter. You can use `make format` to do this.

Reply:
done